From horms  Fri Mar  7 01:10:23 2008
Return-Path: <debbugs@rietz.debian.org>
X-Spam-Checker-Version: SpamAssassin 3.2.1-kirsty.vergenet.net_2007012401
	(2007-05-02) on kirsty.vergenet.net
X-Spam-Level: 
X-Spam-Status: No, score=-2.6 required=5.0 tests=BAYES_00 autolearn=ham
	version=3.2.1-kirsty.vergenet.net_2007012401
X-Original-To: horms@verge.net.au
Delivered-To: horms@vergenet.net
Received: from mail.au.vergenet.net [202.4.237.240]
	by tabatha.lab.ultramonkey.org with POP3 (fetchmail-6.3.8)
	for <horms@localhost> (single-drop); Fri, 07 Mar 2008 10:10:23 +0900 (JST)
Received: from rietz.debian.org (rietz.debian.org [140.211.166.43])
	by kirsty.vergenet.net (Postfix) with ESMTP id B7D182408E
	for <horms@verge.net.au>; Fri,  7 Mar 2008 09:42:16 +1100 (EST)
Received: from debbugs by rietz.debian.org with local (Exim 4.63)
	(envelope-from <debbugs@rietz.debian.org>)
	id 1JXOjw-00088q-Ps; Thu, 06 Mar 2008 22:39:04 +0000
X-Loop: owner@bugs.debian.org
Subject: Bug#469764: ldirectord: Please add LSB formatted dependency info in init.d script
Reply-To: Morten Werner Forsbring <werner@debian.org>, 469764@bugs.debian.org
Resent-From: Morten Werner Forsbring <werner@debian.org>
Resent-To: debian-bugs-dist@lists.debian.org
Resent-CC: Simon Horman <horms@debian.org>
Resent-Date: Thu, 06 Mar 2008 22:39:02 +0000
Resent-Message-ID: <handler.469764.B.120484301829524@bugs.debian.org>
X-Debian-PR-Message: report 469764
X-Debian-PR-Package: ldirectord
X-Debian-PR-Keywords: patch
X-Debian-PR-Source: heartbeat
Received: via spool by submit@bugs.debian.org id=B.120484301829524
          (code B ref -1); Thu, 06 Mar 2008 22:39:02 +0000
Received: (at submit) by bugs.debian.org; 6 Mar 2008 22:36:58 +0000
Received: from mail-forward.uio.no ([129.240.10.42])
	by rietz.debian.org with esmtp (Exim 4.63)
	(envelope-from <werner@forsbring.no>)
	id 1JXOht-0007fX-KL
	for submit@bugs.debian.org; Thu, 06 Mar 2008 22:36:58 +0000
Received: from mail-mx3.uio.no ([129.240.10.44])
	by pat.uio.no with esmtp (Exim 4.67)
	(envelope-from <werner@forsbring.no>)
	id 1JXOhn-0000kO-Pq
	for submit@bugs.debian.org; Thu, 06 Mar 2008 23:36:51 +0100
Received: from sid.forsbring.no ([158.36.191.200])
	by mail-mx3.uio.no with esmtps (TLSv1:AES256-SHA:256)
	(Exim 4.67)
	(envelope-from <werner@forsbring.no>)
	id 1JXOhn-0001g2-Bn
	for submit@bugs.debian.org; Thu, 06 Mar 2008 23:36:51 +0100
Received: from localhost
	([127.0.0.1] helo=sid.forsbring.no ident=werner)
	by sid.forsbring.no with esmtp (Exim 4.63)
	(envelope-from <werner@forsbring.no>)
	id 1JXPW1-0005VV-Vo
	for submit@bugs.debian.org; Fri, 07 Mar 2008 00:28:46 +0100
From: Morten Werner Forsbring <werner@debian.org>
To: submit@bugs.debian.org
Date: Fri, 07 Mar 2008 00:28:45 +0100
Message-ID: <87skz35t5e.fsf@sid.forsbring.no>
User-Agent: Gnus/5.110006 (No Gnus v0.6) Emacs/23.0.60 (gnu/linux)
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
X-UiO-Spam-info: not spam, SpamAssassin (score=0.0, required=5.0, autolearn=disabled, none)
X-UiO-Scanned: 0B0CA6F9839D49CEBA45966491A982F45BE308F3
X-UiO-SR-test: EE6083E1527812C73ED972CFAF5726B00DA42D2A
X-UiO-SPAM-Test: remote_host: 158.36.191.200 spam_score: 0 maxlevel 200 minaction 2 bait 0 mail/h: 4 total 287 max/h 16 blacklist 0 greylist 0 ratelimit 0
Delivered-To: submit@bugs.debian.org
Resent-Sender: Debian BTS <debbugs@rietz.debian.org>
Resent-Date: Thu, 06 Mar 2008 22:39:04 +0000
X-Loop: oathstix_finish by kirsty.vergenet.net
X-Loop: oathstix_finish by tabatha.lab.ultramonkey.org
Status: RO
Content-Length: 2365
Lines: 70

Package:  ldirectord
Version:  2.1.3-3
Tags:     patch
User:     initscripts-ng-devel@lists.alioth.debian.org
Usertags: missing-dependency

To be able to check boot script order, and also to be able to start
boot scripts in parallel, it is important to know the dependencies of
the various boot scripts.  The Linux Software Base specifies a init.d
header file format useful for this purpose, and adding such header to
the ldirectord init.d script would make it possible for us to use this
information to check the current sequence and to speed up the Debian
boot.

I am helping out with a system to update the boot sequence based on
these dependencies, and would like see this as the default in Lenny.
Because of this, it is nice if the dependencies was updated quickly.

<URL:http://refspecs.freestandards.org/LSB_2.1.0/LSB-generic/LSB-generic/initscrcomconv.html>
documents the LSB header format.  Some Debian notes are available from
<URL:http://wiki.debian.org/LSBInitScripts>.

Here is a patch to document the dependencies.  I hope this is correct.

Thanks in advance.


- Werner

diff -ruN heartbeat-2.1.3.orig/ldirectord/init.d/ldirectord.debian heartbeat-2.1.3/ldirectord/init.d/ldirectord.debian
--- heartbeat-2.1.3.orig/ldirectord/init.d/ldirectord.debian 2007-12-21 21:45:48.000000000 +0100
+++ heartbeat-2.1.3/ldirectord/init.d/ldirectord.debian 2008-03-06 23:33:32.864014191 +0100
@@ -1,4 +1,13 @@
 #!/bin/sh
+#
+### BEGIN INIT INFO
+# Provides:          ldirectord
+# Required-Start:    $remote_fs $syslog
+# Required-Stop:     $remote_fs $syslog
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+### END INIT INFO
+#
 # Author: Horms <horms@verge.net.au>
 #
 # Debian init script for ldirectord
diff -ruN heartbeat-2.1.3.orig/ldirectord/init.d/ldirectord.debian.in
heartbeat-2.1.3/ldirectord/init.d/ldirectord.debian.in
--- heartbeat-2.1.3.orig/ldirectord/init.d/ldirectord.debian.in
2007-12-21 16:32:27.000000000 +0100
+++ heartbeat-2.1.3/ldirectord/init.d/ldirectord.debian.in
2008-03-06 23:33:50.815012186 +0100
@@ -1,4 +1,13 @@
 #!/bin/sh
+#
+### BEGIN INIT INFO
+# Provides:          ldirectord
+# Required-Start:    $remote_fs $syslog
+# Required-Stop:     $remote_fs $syslog
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+### END INIT INFO
+#
 # Author: Horms <horms@verge.net.au>
 #
 # Debian init script for ldirectord

